### DISCUSSION TITLE
Renew manually

### DISCUSSION THREAD

**Discussion by raph91** [polymind-inc/acmebot#207] [Q&A] [answered]
Hi

First of all, thank you for distributing this acmebot. It seems to work like a charm.
I was looking for a possibility to test the automatic renewal process. Is it possible to renew a certificate manually?

Thank you.

BR raph

**Comment by shibayan** [comment, marked as answer]
If you want to force a manual renewal, you can do so via "/renew-certificate".

There is currently no way to test the timer execution. But, the internal implementation is identical to the one that is executed periodically by a timer, so if it works here, it should be fine.

**Reply by raph91** [reply]
Thank you for the feedback! Maybe that would be a thing for the documentation.

Is it possible to change the renewal timer? It could be the case to renew the certificates every 30 days. According to the readme the certificates will be renewed after 60 days, 30 days before expiration.

**Reply by shibayan** [reply]
The update timer is supposed to run three times per week, but the actual update is fixed at 30 days before the expiration date. There are no plans to make this 30-day setting changeable at this time to avoid throttling or caching effects from the change.

Note that Let's Encrypt cache the results of the Challenge for 30 days, so testing by changing the update timing is not very effective.

**Comment by sandromastronardi** [comment]
I have a question about this renewal.  The new certificate (manual renewal) gets a new thumbprint... but it keeps using the old certificate for the app service, even after I restart it... when will it use the new one? is that at the end of the final certificate? (so 90 days)? and so it switches 30 days after the renewal?

**Reply by shibayan** [reply]
That's the App Service side of things, so read this document.

https://docs.microsoft.com/en-us/azure/app-service/configure-ssl-certificate

**Reply by sandromastronardi** [reply]
Ah, thanks...
I found the answer there... it can take 48 hours:

If you update your certificate in Key Vault with a new certificate, App Service automatically syncs your certificate within 48 hours.

**Reply by shibayan** [reply]
If you are in a hurry, go to Azure Portal and select the certificate you imported from Key Vault, and you should see a sync button.
![image](https://user-images.githubusercontent.com/1356444/108215792-0ee60500-7175-11eb-857f-eb37f3c65583.png)

**Reply by sandromastronardi** [reply]
Thanks, I already found that one :)
I wasn't in a hurry but wanted to get a 100% insight into the workings before suggesting this for my client :)
Thanks for the quick help

**Comment by jgross-OBA** [comment]
Hi, we have deployed this to Azure and were successfully able to provision new certificates as well as to manually renew a certificate.

The question that we have is that, although we know that LetsEncrypt will not allow us to renew a certificate early, we wanted to know if there is a way to know that keyvault-acmebot would actually attempt to renew as we get to the > 30 day's. We could wait but would prefer to know ahead of that so we can actually move to production as if we ended up having to renew manually it would not be optimal.

I know it may sound like we are being overly paranoid but would prefer to validate this before we go to production.

Thanks

**Reply by shibayan** [reply]
There is no way to automatically renew a certificate more than 30 days before it expires.

However, you can set Key Vault's Issuance Policy to notify you at any time before the certificate expires. Our recommendation is to set up a webhook for Slack or Teams. You will be notified when a certificate is successfully issued or when it fails.

**Comment by jgross-OBA** [comment]
Thank you.

One more question. We had a certificate in our keyvault that had expired a few months ago. When we brought up keyvault-acmebot and pointed it to that keyvault it did not seem to automatically attempt to renew that cert. Is it because it had expired already or was it because it was created outside of the automation? 

Thanks again

**Reply by shibayan** [reply]
Key Vault Acmebot uses the value of the tag attached to the certificate to check if it was issued by itself.

![image](https://user-images.githubusercontent.com/1356444/117152982-7ebf6f80-adf5-11eb-8492-71592cb9665e.png)

Therefore, automatic renewal will not be performed manually or for certificates issued by another CA.
